cardano-cli/src/Cardano/CLI/Conway/Commands.hs

```diff
@@ -35,16 +43,40 @@ runGovernanceCreateVoteCmd
   -> VoteChoice
   -> VoterType
   -> TxIn
-  -> StakeIdentifier
+  -> VerificationKeyOrFile StakePoolKey
   -> ConwayVoteFile Out
   -> ExceptT GovernanceCmdError IO ()
-runGovernanceCreateVoteCmd (AnyShelleyBasedEra sbe) vChoice cType govActionTxIn votingStakeCred oFp = do
-  stakeCred <- firstExceptT StakeCredGovCmdError $ getStakeCredentialFromIdentifier votingStakeCred
-  votingCred <- hoistEither $ first VotingCredentialDecodeGovCmdEror $ toVotingCredential sbe stakeCred
+runGovernanceCreateVoteCmd (AnyShelleyBasedEra sbe) vChoice cType govActionTxIn stakePoolVerKeyOrFile oFp = do
+  stakePoolVerKey <- firstExceptT VotingCredentialDecodeError
+    . newExceptT
+    $ readVerificationKeyOrFile AsStakePoolKey stakePoolVerKeyOrFile
+
+  let stakePoolVerificationKeyHash = unStakePoolKeyHash $ verificationKeyHash stakePoolVerKey
+  votingCred <- hoistEither $ first VotingCredentialDecodeGovCmdEror $ eraDecodeVotingCredential sbe $ Plain.serialize $ Ledger.KeyHashObj stakePoolVerificationKeyHash
```

Comment:
I wasn't able to make changes in `getStakeCredentialFromVerifier`:
* this would require changing `StakeVerifier` type
  * the type is mapped down the road to `Cardano.Api.Address.StakeCredentialByKey`
    * this requires extending json instances
    * and is isomorphic to `StakeCredential StandardCrypto` from Ledger
      * `type StakeCredential c = Credential 'Staking c` - this means I can't have `Credential StakePool StandardCrypto`
 
In the end, the domino effect felt like I'm going into the wrong direction.

This here is too low-level, and I plan to reuse api functions here, but I wasn't able to unify this logic with `getStakeCredentialFromVerifier` easily.

Reply:
It's not the wrong direction because we also want DReps to also be able to cast votes and they won't be using stake pool keys. However, in the interest of time this is ok. Leave a "-- TODO: Conway Era - Use StakeIdentifier"